### PR TITLE
Setup Rubocop for this template (not for the generated apps)

### LINKED ISSUE #146: Should we create a rubocop config for this template repo (not for the generated apps)
It confuses new devs to this project that rubocop doesn't pass on this repo. We would probably need a custom `.rubocop.yml` because quite a bit of the "railsy" one we normally use doesn't apply e.g. any of the `Rails/` or `Performance/` cops.

### PR DESCRIPTION
**Pull request by eoinkelly**
Just some janitorial work to bring this repo into line with our usual ruby code linting. Closes #146.

### CODE REVIEW

File: .rubocop.yml
Comment (eoinkelly) [inline .rubocop.yml:38]: Highlighting this change because it might warrant some discussion. We are using a global var for the config instance to save passing it around. I guess we could refactor it to something nice but I went with the smallest change possible.

File: config/initializers/check_env.rb
Comment (robotdana) [inline config/initializers/check_env.rb:7]: i messed up here and deleted the closing `` ` ``

🐘

File: config/initializers/content_security_policy.rb
Comment (robotdana) [inline config/initializers/content_security_policy.rb:115]: can you remove the arg entirely here? or is it required

File: spec/rails_helper.rb
Comment (robotdana) [inline spec/rails_helper.rb:26]: :heart:

File: variants/bullet/template.rb
Comment (robotdana) [inline variants/bullet/template.rb:2]: maybe a `<<~` should go here? 🐘

File: .rubocop.yml
Comment (robotdana) [inline .rubocop.yml:1]: might pay to have a comment at the top of this file that this is only for this repo, not for the template?

**Review by robotdana — APPROVED** [review]

File: .rubocop.yml
Comment (robotdana) [reply .rubocop.yml:38]: i'd maybe prefer disabling in place, just to discourage any other global vars showing up?

File: config/initializers/content_security_policy.rb
Comment (eoinkelly) [reply config/initializers/content_security_policy.rb:115]: I think keeping the arg helps as documentation so future dev doesn't have to go lookup the docs to see that the request is passed in. That being said, I'm happy to go with what the consensus of reviews is here.

File: variants/bullet/template.rb
Comment (eoinkelly) [reply variants/bullet/template.rb:2]: Some of the usage of `<<-` is deliberate to achive a particular indentation. If all the lines are the same indentation then `<<~` will remove all prefixed whitespace.

Some of the usage is just because `<<~` didn't exist when that code was written so could probably be upgraded. It's a small enough win that I'm not keen to spend time on the effort.

File: .rubocop.yml
Comment (eoinkelly) [reply .rubocop.yml:38]: It ended up a bit ugly because the Rubocop disabling comments were required in some of the generated code. Instead I have replaced it with a constant called `TEMPLATE_CONFIG`. Using a constant like this doesn't feel super idiomatic but it seems equivalent to what we had before and it keeps Rubocop happy :shrug:

Comment (G-Rath) [reply .rubocop.yml]: fwiw you can also configure Rubocop to allow specific globals: https://docs.rubocop.org/rubocop/cops_style.html#configurable-attributes-35

File: config/initializers/content_security_policy.rb
Comment (G-Rath) [reply config/initializers/content_security_policy.rb:115]: I'm not that fussed either, but if I had to pick I'd favor removing it because I don't think it's likely we're going to ever write our own nonce generator

**Review by G-Rath — APPROVED** [review]

File: README.md
Comment (eoinkelly) [inline README.md:108]: @G-Rath I missed in my review of your shakapacker PR that we should update the README. It's a trivial change so I'm just doing it here rather than a separate PR.

**Comment by eoinkelly** [comment]
This change isn't controversial so I'm going to merge it without waiting for other reviews from busy people.

**Pull request merged**
